core/trino-main/src/main/java/io/trino/server/GracefulShutdownHandler.java

```diff
@@ -83,13 +83,10 @@ public synchronized void requestShutdown()
         }
         shutdownRequested = true;
 
-        // wait for a grace period to start the shutdown sequence
+        // wait for a grace period (so that shutting down state is observed by the coordinator) to start the shutdown sequence
         shutdownHandler.schedule(() -> {
```

Comment:
I think it would make it more obvious that a delay is present before the handler runs by extracting the lambda into a method reference.